frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx

```diff
@@ -0,0 +1,387 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import { Helmet } from 'react-helmet';
+import { ActionGroup, Button } from '@patternfly/react-core';
+
+import { ButtonBar, Dropdown, Firehose, history, StatusBox } from '../../utils';
+import {
+  getAlertManagerConfig,
+  patchAlertManagerConfig,
+  receiverTypes,
+} from '../alert-manager-utils';
+import { K8sResourceKind } from '../../../module/k8s';
+import {
+  AlertManagerConfig,
+  AlertManagerReceiver,
+  AlertManagerRoute,
+} from '../alert-manager-config';
+import { RoutingLabelEditor } from './routing-labels-editor';
+import * as PagerDutyForm from './pagerduty-receiver-form';
+import * as WebhookForm from './webhook-receiver-form';
+
+/**
+ * Converts routes of a specific Receiver:
+ * {
+ *   receiver: "MyReceiver",
+ *   match: {
+ *     severity: "warning",
+ *     cluster: "myCluster"
+ *   },
+ *   match_re: {
+ *    service: "$foobar"
+ *  }
+};
+ * ...to array of labels for Routing Labels Editor component
+ * [
+ *   {
+ *     "name": "severity",
+ *     "value": "warning",
+ *     "isRegex": false
+ *   },
+ *   {
+ *     "name": "cluster",
+ *     "value": "myCluster",
+ *     "isRegex": false
+ *   },
+ *   {
+ *     "name": "service",
+ *     "value": "$foobar",
+ *     "isRegex": true
+ *   }
+ * ]
+ */
+const getReceiverRouteLabels = (receiver: AlertManagerReceiver, routes: AlertManagerRoute[]) => {
```

Comment:
You should define TypeScript types for the labels

Reply:
Hi @spadgett, do you mean for the return type?  I now have:
```
const convertReceiverRoutesToEditorLabels = (
  receiver: AlertmanagerReceiver,
  routes: AlertmanagerRoute[],
): RouterEditorLabels => { ...
```
I tried doing things like `const convertReceiverRoutesToEditorLabels: ReceiverRouteLabelProps = ...` or `const convertReceiverRoutesToEditorLabels: React.FC<ReceiverRouteLabelProps> = ...`  but I keep getting typescript errors which I'm investigating.  It shouldn't be `React.FC<...>` as I'm not returning jsx, right? -thanks

Reply:
fixed